src/main/java/org/jenkinsci/plugins/workflow/pipelinegraphanalysis/NodeUtils.java

```diff
@@ -0,0 +1,40 @@
+package org.jenkinsci.plugins.workflow.pipelinegraphanalysis;
+
+import hudson.model.Action;
+import org.jenkinsci.plugins.pipeline.SyntheticStage;
+import org.jenkinsci.plugins.workflow.actions.LabelAction;
+import org.jenkinsci.plugins.workflow.actions.StageAction;
+import org.jenkinsci.plugins.workflow.actions.TagsAction;
+import org.jenkinsci.plugins.workflow.actions.ThreadNameAction;
+import org.jenkinsci.plugins.workflow.graph.FlowNode;
+
+import javax.annotation.CheckForNull;
+import javax.annotation.Nullable;
+
+/**
+ * @author cliffmeyers
+ */
+public class NodeUtils {
```

Comment:
The methods in this class are borrowed from `blueocean-pipeline-api-impl` `PipelineNodeUtil` class. Given the subtly of the logic it seems important to include.. but perhaps this should be placed in some other library?

Reply:
I think you could probably just look for the TagsAction being present with the specific hardcoded string needed -- eliminating the need for this.

Reply:
I simplified this as you suggested in 08b0e6c